master/internal/trial.go

```diff
@@ -899,9 +935,11 @@ func (t *trial) processContainerTerminated(
 	// Terminate the task if the container never started (since this prevents the gang
 	// from ever being able to start), if the leader of the gang has exited out, or if
 	// one of the containers exited with a failure.
-	if c == nil || c.IsLeader() || msg.ContainerStopped.Failure != nil {
+	if !ok || t.containerOrdinals[msg.Container.ID] == 0 || msg.ContainerStopped.Failure != nil {
 		t.terminate(ctx, true)
 	}
+
+	t.terminated(ctx)
```

Comment:
blocking: I don't think we are handling distributed scenarios correctly here because right now we are are terminating the trial as soon as one of the containers exits out.

Reply:
fixed it.